sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/ShowTablesExec.scala

```diff
@@ -0,0 +1,59 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.execution.datasources.v2
+
+import scala.collection.mutable.ArrayBuffer
+
+import org.apache.spark.rdd.RDD
+import org.apache.spark.sql.catalog.v2.{Identifier, TableCatalog}
+import org.apache.spark.sql.catalog.v2.CatalogV2Implicits.NamespaceHelper
+import org.apache.spark.sql.catalyst.InternalRow
+import org.apache.spark.sql.catalyst.encoders.RowEncoder
+import org.apache.spark.sql.catalyst.expressions.{AttributeReference, GenericRowWithSchema}
+import org.apache.spark.sql.catalyst.plans.ShowTablesSchema
+import org.apache.spark.sql.catalyst.util.StringUtils
+import org.apache.spark.sql.execution.LeafExecNode
+
+/**
+ * Physical plan node for showing tables.
+ */
+case class ShowTablesExec(catalog: TableCatalog, ident: Identifier, pattern: Option[String])
+  extends LeafExecNode {
+  override def output: Seq[AttributeReference] = ShowTablesSchema.attributes()
```

Comment:
Exec nodes should use the same attributes as the logical plan nodes they are created from. I think this should have an additional param for `output: Seq[Attribute]`.

Reply:
Then, this needs to be updated as well? https://github.com/apache/spark/pull/25040/files#diff-2e91a80e6e602695acbfe626b54eba50R34

Reply:
Fixed this and `ShowTablesStatement` -> `ShowTables` as well. Now the `explain` displays the plans correctly (before, outputs were not displayed because of the mismatches):
```
== Analyzed Logical Plan ==
database: string, tableName: string, isTemporary: boolean
ShowTables [database#19, tableName#20, isTemporary#21], org.apache.spark.sql.sources.v2.TestInMemoryTableCatalog@3ce7394f, `db`

== Optimized Logical Plan ==
ShowTables [database#19, tableName#20, isTemporary#21], org.apache.spark.sql.sources.v2.TestInMemoryTableCatalog@3ce7394f, `db`

== Physical Plan ==
ShowTables [database#19, tableName#20, isTemporary#21], org.apache.spark.sql.sources.v2.TestInMemoryTableCatalog@3ce7394f, `db`
```

Reply:
Fixed this and `ShowTablesStatement` -> `ShowTables` as well. Now the expression ids are the same across the plans:
```
== Analyzed Logical Plan ==
database: string, tableName: string, isTemporary: boolean
ShowTables [database#19, tableName#20, isTemporary#21], org.apache.spark.sql.sources.v2.TestInMemoryTableCatalog@3ce7394f, `db`

== Optimized Logical Plan ==
ShowTables [database#19, tableName#20, isTemporary#21], org.apache.spark.sql.sources.v2.TestInMemoryTableCatalog@3ce7394f, `db`

== Physical Plan ==
ShowTables [database#19, tableName#20, isTemporary#21], org.apache.spark.sql.sources.v2.TestInMemoryTableCatalog@3ce7394f, `db`
```